src/org/zaproxy/zap/extension/httpsinfo/MDialog.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.httpsinfo;
+
+import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Toolkit;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JTextArea;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.view.AbstractFrame;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+
+public class MDialog extends AbstractFrame {
+	private static final long serialVersionUID = 6633774164518653658L;
+
+	private static final String NEWLINE = "\n";
+	
+	private static final int BEAST_PLUGIN_ID=10200;
+	private static final int CRIME_PLUGIN_ID=10201;
+
+	private ExtensionAlert extensionAlert = (ExtensionAlert) Control.getSingleton().getExtensionLoader()
+			.getExtension(ExtensionAlert.NAME);
+
+	boolean ordered = false;
+	Map<String, Integer> versionStrings;
+	SSLServer server;
+	HttpMessage baseMessage;
+
+	JTextArea general;
+	JLabel spinnerPref;
+	JTextArea cipherSuites;
+	JComboBox<String> suppVers;
+	JButton refresh;
```

Comment:
I'm glad to fix all these things, but know that many of them were inherited from the original code.